NumLib/Fem/FiniteElement/TemplateIsoparametric.h

```diff
@@ -101,13 +103,40 @@ class TemplateIsoparametric
      */
     template <ShapeMatrixType T_SHAPE_MATRIX_TYPE>
     void computeShapeFunctions(const double* natural_pt, ShapeMatrices& shape,
-                               const unsigned global_dim) const
+                               const unsigned global_dim,
+                               bool is_axially_symmetric) const
     {
         NaturalCoordsMappingType::template computeShapeMatrices<
             T_SHAPE_MATRIX_TYPE>(*_ele, natural_pt, shape, global_dim);
+        computeIntegralMeasure(is_axially_symmetric, shape);
+    }
+
+    double interpolateZerothCoordinate(
+        typename ShapeMatrices::ShapeType const& N) const
+    {
+        auto* nodes = _ele->getNodes();
+        typename ShapeMatrices::ShapeType rs(N.size());
+        for (int i=0; i<rs.size(); ++i) {
+            rs[i] = (*nodes[i])[0];
+        }
+        auto const r = N.dot(rs);
+        return r;
```

Comment:
For the general case of Neumann BC, 2 pi* int q(l) r \phi dl is the integration formulas. However, if the Neumann BC is applied to the axis, where r = 0,  the correct formulas is
int q(z) \phi  dz. To avoid changing Neumann BC integration function, I suggested to added this if-condition.